packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -984,6 +1049,59 @@ class MediaQuery extends InheritedModel<_MediaQueryAspect> {
     );
   }
 
+  /// Wraps the `child` in a [MediaQuery] with its [MediaQueryData.textScaler]
+  /// set to [TextScaler.noScaling].
+  ///
+  /// The returned widget must be inserted in a widget tree below an existing
+  /// [MediaQuery] widget.
+  ///
+  /// This can be used to prevent, for example, icon fonts from scaling as the
+  /// user adjusts the platform's text scaling value.
+  static Widget disableTextScaling({
```

Comment:
Uh what would be the benefits for making this a factory constructor? These are 2 new methods so there's no backward compatibility issues (i.e. they don't have to return a subtype of `MediaQuery`). To subclass `MediaQuery` I think it requires a separate Element implementation too.

Reply:
Yeah you're right, it doesn't make sense as a factory constructor.

Whenever I see a static method that returns a Widget I always double check that it couldn't be its own StatelessWidget instead.  In this case it could and it would arguably simplify the internals, but maybe the loss of organization makes it not worth it.  MediaQuery.disableTextScaling looks a lot better than MediaQueryWithNoTextScaling or whatever.  I'm fine keeping it as-is.